src/main/kotlin/Messenger.kt

```
@@ -0,0 +1,151 @@
+package chattore
+
+import chattore.feature.DiscordBroadcastEvent
+import chattore.feature.SpyEnabled
+import com.velocitypowered.api.proxy.Player
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.jsonArray
+import kotlinx.serialization.json.jsonObject
+import kotlinx.serialization.json.jsonPrimitive
+import net.kyori.adventure.text.Component
+import net.kyori.adventure.text.TextReplacementConfig
+import net.kyori.adventure.text.serializer.plain.PlainTextComponentSerializer
+import java.net.URL
+import java.util.*
+
+class Messenger(
+    private val plugin: ChattORE,
+    val format: String
+) {
+    private var fileTypeMap: Map<String, List<String>> = hashMapOf()
+    init {
+        plugin.javaClass.getResourceAsStream("/filetypes.json")?.let { inputStream ->
```

`fileTypeMap` can be a val. Instead of ignoring null, we should throw here because that probably indicates a bug (`filetypes.json` missing somehow)
I think the stream and reader should be closed also (later we can think of a function that does the loading business since it's used elsewhere too)